sql/org_mozilla_firefox_derived/baseline_daily_v1/query.sql

```diff
@@ -0,0 +1,152 @@
+/*
+
+This is a daily aggregation of just baseline pings from across the various
+apps that correspond to the different channels of "Fenix", the new Firefox for Android.
+
+*/
+-- The schemas for the different apps differ slightly such that we can't
+-- blindly union over the baseline views; instead, we extract out the set of
+-- fields we want via a UDF before we union.
+CREATE TEMP FUNCTION extract_fields(baseline ANY TYPE) AS (
+  (
+    SELECT AS STRUCT
+      baseline.submission_timestamp,
+      DATE(baseline.submission_timestamp) AS submission_date,
+      LOWER(baseline.client_info.client_id) AS client_id,
+      baseline.sample_id,
+      SAFE.PARSE_DATE('%F', SUBSTR(baseline.client_info.first_run_date, 1, 10)) AS first_run_date,
+      baseline.ping_info.parsed_end_time AS end_time,
+      udf.glean_timespan_seconds(baseline.metrics.timespan.glean_baseline_duration) AS duration,
+      baseline.client_info.android_sdk_version,
+      baseline.client_info.app_build,
+      baseline.client_info.app_channel,
+      baseline.client_info.app_display_version,
+      baseline.client_info.architecture,
+      baseline.client_info.device_manufacturer,
+      baseline.client_info.device_model,
+      baseline.client_info.telemetry_sdk_build,
+      baseline.client_info.locale,
+      baseline.metadata.geo.city,
+      baseline.metadata.geo.country,
+      baseline.metrics.string.glean_baseline_locale,
+      baseline.normalized_os,
+      baseline.normalized_os_version,
+  )
+);
+
+WITH base AS (
+  SELECT
+    extract_fields(baseline).*,
+    'release' AS normalized_channel
+  FROM
+    org_mozilla_firefox.baseline AS baseline
+  UNION ALL
+  SELECT
+    extract_fields(baseline).*,
+    'beta' AS normalized_channel
+  FROM
+    org_mozilla_firefox_beta.baseline AS baseline
+  UNION ALL
+  SELECT
+    extract_fields(baseline).*,
+    'nightly' AS normalized_channel
+  FROM
+    org_mozilla_fenix.baseline AS baseline
+  -- These final two apps should be retired in February.
+  UNION ALL
+  SELECT
+    extract_fields(baseline).*,
+    'aurora nightly' AS normalized_channel
+  FROM
+    org_mozilla_fennec_aurora.baseline AS baseline
+  UNION ALL
+  SELECT
+    extract_fields(baseline).*,
+    'preview nightly' AS normalized_channel
+  FROM
+    org_mozilla_fenix_nightly.baseline AS baseline
+),
+--
+windowed AS (
+  SELECT
+    submission_date,
+    client_id,
+    sample_id,
+    ROW_NUMBER() OVER w1_unframed AS _n,
+    --
+    -- Take the earliest first_run_date if ambiguous.
+    MIN(first_run_date) OVER w1 AS first_run_date,
+    --
+    -- Sums over distinct baseline pings.
+    SUM(IF(duration BETWEEN 0 AND 100000, duration, 0)) OVER w1 AS durations,
+    --
+    -- Maintain lists of unique client activity dates for usage criteria based on client timestamps.
+    ARRAY_AGG(DATE(TIMESTAMP_SUB(end_time, INTERVAL duration SECOND))) OVER w1 AS start_dates,
```

Comment:
should this use `IF(duration BETWEEN 0 AND 100000, duration, NULL)` for both of these cases, so that it drops start dates from to oversize durations?